src/test/java/com/rakovets/course/java/core/practice/string/StringUtilTest.java

```diff
@@ -0,0 +1,289 @@
+    @MethodSource("punctuations")
+    void punctuations(int expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        int actualString = StringUtil.punctuations(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>palindrome() {
+        return Stream.of(
+                Arguments.of("true", "deed"),
+                Arguments.of("false", "deeds"),
+                Arguments.of("true", "1991")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("palindrome")
+    void palindrome(boolean expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        boolean actualString = StringUtil.palindrome(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>samePartsString() {
+        return Stream.of(
+                Arguments.of(new String[] {"Hel", "lo ", "Jav", "a!"}, "Hello Java!", "3"),
+                Arguments.of(new String[] {"Hello", " Java", "!"}, "Hello Java!", "5")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("samePartsString")
+    void samePartsString(String[] expectedSymbol, String str1, int i) {
+        // GIVEN
+
+        // WHEN
+        String[] actualString = StringUtil.samePartsString(str1, i);
+
+        // THAT
+        Assertions.assertArrayEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>wordCount() {
+        return Stream.of(
+                Arguments.of("2", "Hello Java!"),
+                Arguments.of("7", "ajfk afnoiafn iwfn qwjfjn,oiaf. oauhf! Fqwf?"),
+                Arguments.of("1", "1991")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("wordCount")
+    void wordCount(int expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        int actualString = StringUtil.wordCount(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>initialsFullName() {
+        return Stream.of(
+                Arguments.of("EK", "Evgen Kolesnik"),
+                Arguments.of("AA", "ajfk afnoiafn iwfn qwjfjn,oiaf. oauhf! Fqwf?"),
+                Arguments.of("PO", "pasfasf,owfwf")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("initialsFullName")
+    void initialsFullName(String expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.initialsFullName(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>allNumbers() {
+        return Stream.of(
+                Arguments.of("1", "Evgen Kolesn1k"),
+                Arguments.of("1231244", "ajfk afnoiafn iwf123,1244afsf"),
+                Arguments.of("", "pafqf")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("allNumbers")
+    void allNumbers(String expectedSymbol, String str1) {
```

Comment:
"Violation: 'nameMethodTest' was not used for test-method naming
Solution: Rename method
"